dts/riscv/hpmicro/hpm6750.dtsi

```diff
@@ -0,0 +1,178 @@
+		ranges;
+		clk: clock-controller@f4100000 {
+			compatible = "hpmicro,hpm-clock";
+			reg = <0xf4100000 0x100>;
+			clock-frequency = <816000000>;
+			reg-names = "clk";
+			status = "okay";
+			clock-sys-core = <HPM_CLOCK_CPU0>;
+			ram-up-time = <(32*1000*9)>;
+			sysctl-present = "sysctl-preset-0";
+			#clock-cells = <3>;
+		};
+
+		mtimer: timer@e6000000 {
+			compatible = "andestech,machine-timer";
+			reg = <0xe6000000 0x10>;
+			interrupts-extended = <&CPU0_intc 7 &CPU1_intc 7>;
+		};
+
+		flash0: flash@80000000 {
+			compatible = "soc-nv-flash";
+			reg = <0x80000000 DT_SIZE_M(16)>;
+		};
+
+		ilm: memory@0 {
+			compatible = "zephyr,memory-region";
+			reg = <0x0 DT_SIZE_K(256)>;
+			zephyr,memory-region = "ITCM";
+		};
+
+		dlm: memory@80000 {
+			compatible = "zephyr,memory-region";
+			reg = <0x080000 DT_SIZE_K(256)>;
+			zephyr,memory-region = "DTCM";
+		};
+
+		sram: memory@1080000 {
+			device_type = "memory";
+			compatible = "mmio-sram";
+			reg = <0x01080000 DT_SIZE_K(1024)>;
+		};
+
+		core0_lm_slv: memory@1000000 {
+			device_type = "memory";
+			reg = <0x1000000 DT_SIZE_K(512)>;
+		};
+
+		core1_lm_slv: memory@1180000 {
+			device_type = "memory";
+			reg = <0x1180000 DT_SIZE_K(256)>;
+		};
+
+		noncache: memory@11c0000 {
+			device_type = "memory";
+			reg = <0x11c0000 DT_SIZE_K(256)>;
+		};
+
+		plic0: interrupt-controller@e4000000 {
+			compatible = "sifive,plic-1.0.0";
+			#address-cells = <1>;
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			reg = <	0xe4000000 0x00001000
+				0xe4002000 0x00000800
+				0xe4200000 0x00010000 >;
+			reg-names = "prio", "irq_en", "reg";
+			riscv,max-priority = <255>;
+			riscv,ndev = <1023>;
+			interrupts-extended = <&CPU0_intc 0 &CPU1_intc 0>;
+		};
+
+		plic_sw0: interrupt-controller@e6400000 {
+			compatible = "andestech,plic_sw";
+			#address-cells = <1>;
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			reg = <0xe6400000 0x00400000>;
+			riscv,max-priority = <255>;
+			riscv,ndev = <1023>;
+			interrupts-extended = <&CPU0_intc 3 &CPU1_intc 3>;
+		};
+
+		gpio0: peripheral@f0000000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0xf0000000 0x4000>;
+		};
+
+		gpio1: peripheral@f0004000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0xf0004000 0x4000>;
+		};
+
+		uart0: serial@f0040000 {
+			compatible = "hpmicro,hpm-uart";
+			reg = <0xf0040000 0x1000>;
+			interrupts = <31 1>;
+			interrupt-parent = <&plic0>;
+			clocks = <&clk HPM_CLOCK_UART0 HPM_CLK_SRC_OSC24M 1>;
+			status = "disabled";
+		};
+
+		pinctrl: pin-controller@f4040000 {
+			compatible = "hpmicro,hpm-pinctrl";
+			reg = <0xf4040000 0x40000>;
+			status = "okay";
+		};
+		pinctrl_pioc: pin-controller@f40d8000 {
+			compatible = "hpmicro,hpm-pinctrl";
+			reg = <0xf40d8000 0x40000>;
+			status = "okay";
+		};
+
+		pinctrl_bioc: pin-controller@f5010000 {
+			compatible = "hpmicro,hpm-pinctrl";
+			reg = <0xf5010000 0x40000>;
+			status = "okay";
+		};
```

Comment:
please add hw description at the time you add the driver, not before.

Reply:
OK